src/ngx_http_lua_headers.c

```diff
@@ -1298,13 +1298,30 @@ ngx_http_lua_ffi_get_resp_header(ngx_http_request_t *r,
     ngx_uint_t           i;
     ngx_table_elt_t     *h;
     ngx_list_part_t     *part;
+    ngx_http_lua_ctx_t  *ctx;
+    ngx_int_t            rc;
 
     ngx_http_lua_loc_conf_t     *llcf;
 
     if (r->connection->fd == (ngx_socket_t) -1) {
         return NGX_HTTP_LUA_FFI_BAD_CONTEXT;
     }
 
+    ctx = ngx_http_get_module_ctx(r, ngx_http_lua_module);
+    if (ctx == NULL) {
+        // *errmsg = "no ctx found";
+        return NGX_ERROR;
+    }
+
+    if (!ctx->headers_set) {
+        rc = ngx_http_lua_set_content_type(r);
+        if (rc != NGX_OK) {
```

Comment:
We don't really need the `rc` local variable. Just write put the `ngx_http_lua_set_content_type(r)` function call inside the `if` statement condition.